src/uct/api/v2/uct_v2.h

```diff
@@ -0,0 +1,139 @@
+    UCT_OP_EAGER_ZCOPY,  /**< Tag matching zcopy eager */
+    UCT_OP_RNDV_ZCOPY,   /**< Tag matching rendezvous eager */
+    UCT_OP_ATOMIC_POST,  /**< Atomic post */
+    UCT_OP_ATOMIC_FETCH  /**< Atomic fetch */
+} uct_ep_operation_t;
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief UCT interface query by @ref uct_iface_query_perf parameters field mask.
+ *
+ * The enumeration allows specifying which fields in @ref uct_perf_attr_t are
+ * present, for backward compatibility support.
+ */
+enum uct_perf_attr_field {
+    /** Enables @ref uct_perf_attr_t::operation */
+    UCT_PERF_ATTR_FIELD_OPERATION          = UCS_BIT(0),
+
+    /** Enables @ref uct_perf_attr_t::local_memory_type */
+    UCT_PERF_ATTR_FIELD_LOCAL_MEMORY_TYPE  = UCS_BIT(1),
+
+    /** Enables @ref uct_perf_attr_t::remote_memory_type */
+    UCT_PERF_ATTR_FIELD_REMOTE_MEMORY_TYPE = UCS_BIT(2),
+
+    /** Enables @ref uct_perf_attr_t::overhead */
+    UCT_PERF_ATTR_FIELD_OVERHEAD           = UCS_BIT(3),
+
+    /** Enables @ref uct_perf_attr_t::bandwidth */
+    UCT_PERF_ATTR_FIELD_BANDWIDTH          = UCS_BIT(4)
+};
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Parameters for querying a UCT interface by @ref uct_iface_query_perf
+ *
+ * This structure must be allocated and initialized by the user
+ */
+typedef struct {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_perf_attr_field. Fields not specified by this mask will be
+     * ignored. This field must be initialized by the caller.
+     */
+    uint64_t            field_mask;
+
+    /**
+     * Operation to report performance for.
+     * This field must be initialized by the caller.
+     */
+    uct_ep_operation_t  operation;
+
+    /**
+     * Local memory type to use for determining performance.
+     * This field must be initialized by the caller.
+     */
+    ucs_memory_type_t   local_memory_type;
+
+    /**
+     * Remote memory type to use for determining performance.
+     * Relevant only for operations that have remote memory access.
+     * This field must be initialized by the caller.
+     */
+    ucs_memory_type_t   remote_memory_type;
+
+    /**
+     * Message overhead time, in seconds. This field is set by the UCT layer.
+     */
+    double              overhead;
+
+    /**
+     * Bandwidth model. This field is set by the UCT layer.
+     */
+    uct_ppn_bandwidth_t bandwidth;
+} uct_perf_attr_t;
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Get interface performance attributes, by memory types and operation.
+ *        A pointer to uct_perf_attr_t struct must be passed, with the memory
+ *        types and operation members initialized. Overhead and bandwidth
+ *        for the opration on the given memory types will be reported.
+ *
+ * @param [in]    iface     Interface to query.
+ * @param [inout] perf_attr Filled with performance attributes.
+ */
+ucs_status_t
+uct_iface_query_perf(uct_iface_h iface, uct_perf_attr_t *perf_attr);
```

Comment:
@yosefe For consistency reasons, shell we name it uct_iface_query_perf_v2 since the other APIs will have the suffix ?

Reply:
It is not very clear to me how the mask is used in the context of `uct_iface_query_perf()`. Shell the user specify `field_mask` for the fields that users wants to query.

Reply:
> @yosefe For consistency reasons, shell we name it uct_iface_query_perf_v2 since the other APIs will have the suffix ?

IMO, for new functions, no need to have _v2 suffix. It will make transitions easier.

Reply:
@yosefe - IMO, it will be very confusing.

Reply:
As discussed with @yosefe , we'll stick with `uct_iface_query_perf` (other API is `uct_ib_iface_is_roce_v2` which is irrelevant).

Reply:
If we add _v2 suffix to everything now, even when not really needed to avoid name collision, will require extra steps when making these APIs default. We don't want to be stuck for ever with _v2 in the API...